### PR TITLE
Fix canonical path encode

### PR DESCRIPTION
**Pull request by jcbf**
While using Elasticsearch, a came across with this limitation and found that canonical path must be double encoded as described in [RT#126025](https://rt.cpan.org/Public/Bug/Display.html?id=126025).

This was heavely tested in ElasticSearch service.

### CODE REVIEW

File: lib/Net/Amazon/Signature/V4.pm [hunk 1]
```diff
@@ -174,6 +174,16 @@ Maintained by Dan Book, C<< <dbook at cpan.org> >>
 
 =cut
 
+# https://github.com/aws/aws-sdk-java-v2/blob/a9387dcb384ef8d62e022483d7a10635f891d618/utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java#L41
+
+sub _aws_uri_escape {
+	my $elem = uri_escape(shift);
+	$elem =~ s/\+/%20/g;
```
Comment (Grinnz) [inline lib/Net/Amazon/Signature/V4.pm]: Encoding space and * to their percent equivalents and not encoding ~ is the default behavior of uri_escape, so the following "fixups" shouldn't be needed here.

File: lib/Net/Amazon/Signature/V4.pm [hunk 2]
```diff
@@ -174,6 +174,16 @@ Maintained by Dan Book, C<< <dbook at cpan.org> >>
 
 =cut
 
+# https://github.com/aws/aws-sdk-java-v2/blob/a9387dcb384ef8d62e022483d7a10635f891d618/utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java#L41
+
+sub _aws_uri_escape {
+	my $elem = uri_escape(shift);
```
Comment (Grinnz) [inline lib/Net/Amazon/Signature/V4.pm]: This should probably be uri_escape_utf8.

File: lib/Net/Amazon/Signature/V4.pm (hunk 1 above)
Comment (jcbf) [reply lib/Net/Amazon/Signature/V4.pm]: Probably was overzealous.

**Comment by Grinnz** [comment]
With the simple uri_escape does everything still work as expected?

**Comment by jcbf** [comment]
I've tested with official elasticsearch perl client lib and worked with this example

```
my $results = $c->search(
  index => ["index1","index2*","ሴ","foo "],
  ignore_unavailable => 'true',
  body => {
        "sort" => [{"localtime" => { "order"=>"desc"}} ]}
);
```
And that includes  space,comma, * and UTF.

**Comment by Grinnz** [comment]
Thanks!

**Pull request merged**